### PR TITLE
Add EmptyContent to Virtualize

### PR DESCRIPTION
**Pull request by alex-oswald**
# Add EmptyContent to Virtualize component

## Description

This PR adds an `EmptyContent` fragment to the virtualize component. If there are no items to render, and the fragment is not null, the fragment is rendered.

This is useful because as it stands, if there are no items to render, nothing is rendered, potentially leaving a user confused.

Implements: https://github.com/dotnet/aspnetcore/issues/28770

### CODE REVIEW

File: src/Components/Web/src/Virtualization/Virtualize.cs [hunk 1]
```diff
@@ -66,6 +66,12 @@ public sealed class Virtualize<TItem> : ComponentBase, IVirtualizeJsCallbacks, I
     [Parameter]
     public RenderFragment<TItem>? ItemContent { get; set; }
 
+    /// <summary>
+    /// Gets or sets the empty content.
```
Comment (pranavkm) [inline src/Components/Web/src/Virtualization/Virtualize.cs:70]:
```suggestion
    /// Gets or sets the `<see cref="RenderFragment" />` that is used when no items are available.
```

File: src/Components/Web/src/Virtualization/Virtualize.cs [hunk 2]
```diff
@@ -66,6 +66,12 @@ public sealed class Virtualize<TItem> : ComponentBase, IVirtualizeJsCallbacks, I
...
+    /// </summary>
+    [Parameter]
+    public RenderFragment? EmptyContent { get; set; }
```
Comment (pranavkm) [inline src/Components/Web/src/Virtualization/Virtualize.cs:73]: Could you follow the instructions here: https://github.com/dotnet/aspnetcore/blob/main/docs/APIBaselines.md#publicapiunshippedtxt?

File: src/Components/Web/src/Virtualization/Virtualize.cs [hunk 3]
```diff
@@ -193,6 +199,11 @@ protected override void BuildRenderTree(RenderTreeBuilder builder)
 
         builder.OpenRegion(3);
 
+        if (EmptyContent is not null && _itemCount == 0)
```
Comment (pranavkm) [inline src/Components/Web/src/Virtualization/Virtualize.cs:202]: Wouldn't this show the empty content when the list is still being loaded in the event the data source is an `ItemsProvider`? That doesn't seem right.

Could we also add some tests for this?

File: src/Components/Web/src/Virtualization/Virtualize.cs (hunk 2 above)
Comment (alex-oswald) [reply src/Components/Web/src/Virtualization/Virtualize.cs:73]: Yes sir. I missed that part of the instructions.

File: src/Components/Web/src/Virtualization/Virtualize.cs (hunk 3 above)
Comment (alex-oswald) [reply src/Components/Web/src/Virtualization/Virtualize.cs:202]: You're right, it indeed does. I will rectify that and write some tests.

**Comment by aloswald** [comment]
No work plans during the holidays. If you need to close this to keep it clean that’s cool. I can reopen after the holidays when I finish the work.

**Comment by TanayParikh** [comment]
> No work plans during the holidays. If you need to close this to keep it clean that’s cool. I can reopen after the holidays when I finish the work.

No worries, it's just an automated process. Feel free to reopen whenever you're ready 😀

**Pull request closed without merging**